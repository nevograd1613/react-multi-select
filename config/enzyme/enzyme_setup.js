// TODO: Remove this `raf` polyfill once the below issue is sorted
// https://github.com/facebookincubator/create-react-app/issues/3199#issuecomment-332842582
import raf from "./tempPolyfills";

import Enzyme from "enzyme";
import Adapter from "@cfaester/enzyme-adapter-react-18";

Enzyme.configure({ adapter: new Adapter() });
