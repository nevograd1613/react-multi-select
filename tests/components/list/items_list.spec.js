import React from "react";
import ShallowRenderer from "react-shallow-renderer";
import { shallow, mount } from "enzyme";

import ItemsList from "../../../src/components/list/items_list";
import InnerList from "../../../src/components/list/list";

describe("ItemsList", () => {
  test("default snapshot", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<ItemsList width={400} />);
    expect(tree).toMatchSnapshot();
  });

  test("update will trigger forceUpdateGrid", () => {
    const forceUpdateGrid = jest.fn();
    const wrapper = shallow(<ItemsList width={100} />);
    const instance = wrapper.instance();
    instance.listRef = { forceUpdateGrid };
    instance.update();
    expect(forceUpdateGrid).toHaveBeenCalledTimes(1);
  });

  test("componentDidUpdate will trigger update", () => {
    const forceUpdateGrid = jest.fn();
    const wrapper = shallow(<ItemsList width={100} />);
    const instance = wrapper.instance();
    instance.listRef = { forceUpdateGrid };
    instance.componentDidUpdate();
    expect(forceUpdateGrid).toHaveBeenCalledTimes(1);
  });

  test("getlistRef will populate listRef", () => {
    const wrapper = shallow(<ItemsList width={100} />);
    const instance = wrapper.instance();
    instance.getlistRef("test");
    expect(instance.listRef).toBe("test");
  });

  test("calls for inner list", () => {
    const wrapper = mount(<ItemsList width={100} />);
    const innerList = wrapper.find(InnerList);
    expect(innerList.length).toBe(1);
  });

  test("disabledItemsTooltip property injected", () => {
    const wrapper = mount(
      <ItemsList disabledItemsTooltip={"You can select up to 4 items"} />,
    );
    const innerList = wrapper.find(InnerList);
    expect(innerList.prop("disabledItemsTooltip")).toBe(
      "You can select up to 4 items",
    );
  });
});
