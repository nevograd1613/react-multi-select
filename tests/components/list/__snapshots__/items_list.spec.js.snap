// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsList default snapshot 1`] = `
<AutoSizer
  disableHeight={false}
  disableWidth={false}
  onResize={[Function]}
  style={{}}
>
  [Function]
</AutoSizer>
`;
